### PR TITLE
Updates profile api // Adds login gating component

### PR DESCRIPTION
**Pull request by crwallace**
This adds login gating component to `/profile` and `/rx`.

All that is needed is to pass you authRequirement and your component to the `<RequiredLoginView/>` component.

For example: `/profile` only requires an auth level of 1 so I would do the following:

`<RequiredLoginView authRequired={1} component={parentProfileComponent}/>`

@ayaleloehr -- I have already added this component to Rx.
@austin-martinez -- I was going to add this to disability-benefits but I wasn't able to clearly see which component was your parent component. Let me know if you need help implementing this.

### CODE REVIEW

File: config/.sass-lint.yml
Comment (ayaleloehr) [inline config/.sass-lint.yml]: With all the TODOs in this file, what is the "later" that will change it back to 1?  Is there some event or what will happen to allow us to change those to 1?

File: src/js/auth/containers/LoginApp.jsx
Comment (ayaleloehr) [inline src/js/auth/containers/LoginApp.jsx]: remove commented out line

File: src/js/auth/containers/LoginApp.jsx
Comment (ayaleloehr) [inline src/js/auth/containers/LoginApp.jsx]: what if something else is in localStorage, does this remove it? (and is that desired?)

File: src/js/common/components/RequiredLoginView.jsx
Comment (ayaleloehr) [inline src/js/common/components/RequiredLoginView.jsx]: similar to above, what if localStorage has something else in it to make length > 0, but it's not the userToken. Would it be better to check for userToken instead?

File: src/js/common/components/RequiredLoginView.jsx
Comment (ayaleloehr) [inline src/js/common/components/RequiredLoginView.jsx]: remove commented out line

File: src/js/common/components/RequiredLoginView.jsx
Comment (ayaleloehr) [inline src/js/common/components/RequiredLoginView.jsx]: Question not to fix in this PR, but does defining these as specific pixel sizes mess things up on mobile or different screens?  If yes, add a TODO? If no, ignore. :)

File: src/js/common/components/RequiredLoginView.jsx
Comment (ayaleloehr) [inline src/js/common/components/RequiredLoginView.jsx]: It's safe to assume this order will stay the same? (i.e. pulling third will always work?)

File: src/js/common/components/RequiredLoginView.jsx
Comment (ayaleloehr) [inline src/js/common/components/RequiredLoginView.jsx]: Add TODO to replace the FAQ link so we don't forget later?

File: src/js/rx/containers/Main.jsx
Comment (ayaleloehr) [inline src/js/rx/containers/Main.jsx]: one tab left

**Review by ayaleloehr — APPROVED** [review]
Several nits and questions, but overall really nice work here!.  I am absolutely not the best person to review this, so would like to get another set of eyes on it tomorrow before it gets stale in our heads. With that though, I'm good to give this a LGTM and think this can be merged once you feel you have addressed the questions from this review and that one test passes.  

If future reviewer has additional questions, I think addressing in a follow up PR is appropriate (rather than holding up this one).

File: config/.sass-lint.yml
Comment (crwallace) [inline config/.sass-lint.yml]: this file shouldn't even really be here. We aren't using it so I am going to remove it.

File: src/js/auth/containers/LoginApp.jsx
Comment (crwallace) [inline src/js/auth/containers/LoginApp.jsx]: yes it will. No that is not desired.

**Pull request merged**
